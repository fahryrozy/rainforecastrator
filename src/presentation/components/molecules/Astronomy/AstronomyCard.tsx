import {View, Text} from 'react-native';
import React from 'react';
import {styles} from './Astronomy.styles';

const AstronomyCard = ({data}) => {
  return (
    <View style={styles.containerAstro}>
      <View style={styles.astroCard}>
        <Text style={styles.locText}>Sunrise</Text>
        <Text style={styles.locText}>{data?.astronomy?.astro?.sunrise}</Text>
      </View>
      <View style={styles.astroCard}>
        <Text style={styles.locText}>Sunset</Text>
        <Text style={styles.locText}>{data?.astronomy?.astro?.sunset}</Text>
      </View>
      <View style={styles.astroCard}>
        <Text style={styles.locText}>Moonrise</Text>
        <Text style={styles.locText}>{data?.astronomy?.astro?.moonrise}</Text>
      </View>
      <View style={styles.astroCard}>
        <Text style={styles.locText}>Moonset</Text>
        <Text style={styles.locText}>{data?.astronomy?.astro?.moonset}</Text>
      </View>
    </View>
  );
};
export default AstronomyCard;
