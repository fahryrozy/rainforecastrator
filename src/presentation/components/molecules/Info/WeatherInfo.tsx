import {View, Text, Image} from 'react-native';
import React from 'react';
import moment from 'moment';
import 'moment/locale/id';
import {styles} from './CurrentWeather.styles';
import Clock from 'react-live-clock';

const WeatherInfo = ({data}) => {
  moment.locale('id');
  return (
    <View style={styles.mainContainer}>
      {data && (
        <View style={styles.weatherInfoCard}>
          <View style={styles.locCard}>
            <Text ellipsizeMode="tail" numberOfLines={1} style={styles.locText}>
              {data?.location?.name}
            </Text>
            <View style={styles.locTimeCard}>
              <Text style={styles.locTime}>Local time : </Text>
              {data?.location?.tz_id && (
                <Clock
                  style={styles.locTime}
                  format={'HH:mm:ss'}
                  ticking={true}
                  element={Text}
                  timezone={data.location.tz_id}
                />
              )}
            </View>
          </View>

          <View style={styles.tempCard}>
            <View style={styles.iconContainer}>
              <Image
                style={styles.icon}
                source={{
                  uri: `https:${data?.current?.condition?.icon}`,
                }}
              />
              <Text style={styles.tempText}>{`${data?.current?.temp_c}Â°`}</Text>
            </View>

            <Text style={styles.tempInfoText}>
              {data?.current?.condition?.text}
            </Text>

            <Text style={styles.lastUpdDate}>{`Last updated on ${moment(
              data?.current?.last_updated,
              'YYYY-MM-DD H:m',
            ).format('LT')}`}</Text>
          </View>
        </View>
      )}
    </View>
  );
};

export default WeatherInfo;
