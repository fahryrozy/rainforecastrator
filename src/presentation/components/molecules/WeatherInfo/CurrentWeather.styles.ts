import {Colors, scale} from '@core/style';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  weatherInfoCard: {
    flex: 1,
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
    overflow: 'hidden',
    rowGap: scale(2),
  },
  iconContainer: {
    height: scale(75),
    justifyContent: 'center',
    alignItems: 'flex-start',
    flexDirection: 'row',
    width: '100%',
    overflow: 'hidden',
    gap: scale(10),
  },
  locCard: {
    flex: 0.35,
    rowGap: scale(2),
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    overflow: 'hidden',
    paddingVertical: scale(5),
  },
  tempCard: {
    flex: 0.65,
    paddingVertical: scale(2),
    justifyContent: 'flex-end',
    alignItems: 'center',
    width: '100%',
    overflow: 'hidden',
  },
  locTimeCard: {
    flexDirection: 'row',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
    overflow: 'hidden',
  },
  locTime: {
    textAlign: 'center',
    fontSize: scale(10),
    color: Colors.pureWhite,
    fontFamily: 'Poppins',
    fontWeight: '400',
  },
});
