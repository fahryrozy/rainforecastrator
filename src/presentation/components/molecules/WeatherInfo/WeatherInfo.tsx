import {View, Text, Image, ActivityIndicator} from 'react-native';
import React from 'react';
import moment from 'moment';
import 'moment/locale/id';
import {styles} from './CurrentWeather.styles';
import Clock from 'react-live-clock';
import {Current, Location} from '@domain/entities';
import * as Sentry from '@sentry/react';

type Props = {
  data: {
    location: Location;
    current: Current;
  };
  isLoading: boolean;
};

const WeatherInfo: React.FC<Props> = ({data, isLoading}) => {
  moment.locale('id');
  if (isLoading) {
    return (
      <ActivityIndicator
        size="large"
        color="#fff"
        style={styles.mainContainer}
      />
    );
  }
  return (
    <View style={styles.mainContainer}>
      {data && (
        <View style={styles.weatherInfoCard}>
          <View style={styles.locCard}>
            <Text ellipsizeMode="tail" numberOfLines={1} style={styles.locText}>
              {data?.location?.name}
            </Text>
            <View style={styles.locTimeCard}>
              <Text style={styles.locTime}>Local time : </Text>
              {data?.location?.tz_id ? (
                <Clock
                  style={styles.locTime}
                  format={'HH:mm:ss'}
                  ticking={true}
                  element={Text}
                  timezone={data.location.tz_id}
                />
              ) : (
                <Text>N/A</Text>
              )}
            </View>
          </View>

          <View style={styles.tempCard}>
            <View style={styles.iconContainer}>
              <Image
                style={styles.icon}
                source={{
                  uri: `https:${data?.current?.condition?.icon}`,
                }}
              />
              <Text style={styles.tempText}>{`${data?.current?.temp_c}Â°`}</Text>
            </View>

            <Text style={styles.tempInfoText}>
              {data?.current?.condition?.text}
            </Text>

            <Text style={styles.lastUpdDate}>{`Last updated on ${moment(
              data?.current?.last_updated,
              'YYYY-MM-DD H:m',
            ).format('LT')}`}</Text>
          </View>
        </View>
      )}
    </View>
  );
};

export default Sentry.withProfiler(WeatherInfo);
