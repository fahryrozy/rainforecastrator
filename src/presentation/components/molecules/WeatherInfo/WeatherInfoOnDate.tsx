import {View, Text, Image, ActivityIndicator} from 'react-native';
import React from 'react';
import moment from 'moment';
import 'moment/locale/id';
import {styles} from './CurrentWeather.styles';
import Clock from 'react-live-clock';
import {Forecast, Hour, Location} from '@domain/entities';
import {Colors, Typhography} from '@core/style';
import {Wrapper} from '@core/style/Wrapper';
import IconStyle from '@core/style/Icon';

type Props = {
  data:
    | {
        location: Location;
        forecast: Forecast;
      }
    | undefined;
  condition: Hour | undefined;
  isLoading: boolean;
};

const WeatherInfoOnDate: React.FC<Props> = ({data, condition, isLoading}) => {
  moment.locale('en');

  if (isLoading) {
    return <ActivityIndicator size="large" color={Colors.pureWhite} />;
  }
  return (
    <View style={[styles.weatherContainer, Wrapper.ContainerColumn()]}>
      {data && condition && (
        <View style={styles.weatherInfoCard}>
          <View style={styles.locCard}>
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={Typhography.headerBigDefault()}>
              {data?.location?.name}
            </Text>
            <View style={styles.locTimeCard}>
              <Text style={Typhography.bodyDefault()}>Local time : </Text>
              {data?.location?.tz_id ? (
                <Clock
                  style={styles.locTime}
                  format={'HH:mm:ss'}
                  ticking={true}
                  element={Text}
                  timezone={data.location.tz_id}
                />
              ) : (
                <Text style={Typhography.bodyDefault()}>N/A</Text>
              )}
            </View>
          </View>

          <View style={styles.tempCard}>
            <View style={styles.iconContainer}>
              <Image
                style={IconStyle.iconBig}
                source={{
                  uri: `https:${condition.condition?.icon}`,
                }}
              />
              <Text
                style={Typhography.headerExtraBigDefault()}>{`${condition.temp_c}Â°`}</Text>
            </View>

            <Text style={Typhography.headerHalfBigDefault()}>
              {data?.forecast.forecastday[0]?.hour[0]?.condition?.text}
            </Text>

            <Text style={Typhography.bodyBold()}>{`${moment(
              condition.time,
              'YYYY-MM-DD H:m',
            ).format('llll')}`}</Text>
          </View>
        </View>
      )}
    </View>
  );
};

export default WeatherInfoOnDate;
