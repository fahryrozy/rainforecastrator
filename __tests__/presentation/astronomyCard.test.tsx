/**
 * @format
 */

import 'react-native';
import React from 'react';
import {it} from '@jest/globals';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react-native';
import '@testing-library/jest-dom';
import '@testing-library/react-native/extend-expect';
import WeatherInfo from '@presentation/components/molecules/WeatherInfo/WeatherInfo';
import {mockData_astronomy} from '../data/mock';
import AstronomyCard from '@presentation/components/molecules/Astronomy/AstronomyCard';

describe('Weather Info Card in Homescreen', () => {
  it('renders correctly with blank data', () => {
    renderer.create(<WeatherInfo data={undefined} isLoading={false} />);
  });

  it('renders correctly with mock data', () => {
    renderer.create(
      <AstronomyCard data={mockData_astronomy} isLoading={false} />,
    );
  });

  it('match snapshot', () => {
    const snap = renderer
      .create(<AstronomyCard data={mockData_astronomy} isLoading={false} />)
      .toJSON();
    expect(snap).toMatchSnapshot();
  });

  it('has astronomy phase title', () => {
    render(<AstronomyCard data={mockData_astronomy} isLoading={false} />);
    const titleText = screen.getByTestId('as-title');
    expect(titleText).toBeOnTheScreen();
  });

  it('has astronomy phase time', () => {
    render(<AstronomyCard data={mockData_astronomy} isLoading={false} />);
    const titleText = screen.getByTestId('as-time');
    expect(titleText).toBeOnTheScreen();
  });
});
