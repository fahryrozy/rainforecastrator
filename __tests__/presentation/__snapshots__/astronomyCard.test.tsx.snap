// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Info Card in Homescreen match snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderRadius": 20,
      "flexDirection": "row",
      "height": 160,
      "justifyContent": "center",
      "marginVertical": 5,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      Sunrise
    </Text>
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      05:29 AM
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
      testID="as-title"
    >
      Sunset
    </Text>
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      05:57 PM
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      Moonrise
    </Text>
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
      testID="as-time"
    >
      11:44 PM
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "overflow": "hidden",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      Moonset
    </Text>
    <Text
      style={
        {
          "color": "rgba(255, 255, 255, 1)",
          "fontFamily": "Poppins",
          "fontSize": 30,
          "fontWeight": "400",
          "textAlign": "center",
        }
      }
    >
      11:09 AM
    </Text>
  </View>
</View>
`;
